### PR TITLE
Cleanup contributing

### LINKED ISSUE #1378: Align the GOVERNANCE.md and CONTRIBUTING.md documents
The governance model has been approved with this follow-up task:

- move the repository rules from the governance.md to the contributing.md document.

_Originally posted by @TheNeikos in https://github.com/thin-edge/thin-edge.io/pull/1203#discussion_r955893433_

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This is the second part of #1379 and

Closes #1378

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,20 +36,87 @@ When possible, fill out as much detail in the pull request template as is
...
-independently verify and close them.
-
 ## Writing good commit messages
 
-Git commit messages should explain the how and why of your change and be
-separated into a brief subject line followed by a more detailed body. When in
-doubt, follow this guide for good commit messages and you can’t go wrong:
-[https://chris.beams.io/posts/git-commit/](https://chris.beams.io/posts/git-commit/).
+Having clear, concise and well-written commit messages for all changes is not
+only a indicator for the seriousness of the project, but also a way to ensure
```
Comment (TheNeikos) [inline CONTRIBUTING.md:42]:
```suggestion
only an indicator for the seriousness of the project, but also a way to ensure
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -33,20 +36,87 @@ When possible, fill out as much detail in the pull request template as is
...
+individual and separate changes
+* that the commit message of the change expresses why the change was made
+
+Every reviewer of a pull request is asked to review not only the changes, but
+also the commit messages.
+Reviewers are explicitly empowered and actually encouraged to block pull
+requests that contain commit messages that do not meet a certain standard
+even and especially also if the proposed changes are acknowledged.
+
+The contributor should document why they implemented a change properly. A good
```
Comment (TheNeikos) [inline CONTRIBUTING.md:62]:
```suggestion
The contributor should document why they implemented a change. A good
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -33,20 +36,87 @@ When possible, fill out as much detail in the pull request template as is
...
+rule of thumb is that if a change is larger than 60 lines, a commit message
+without a body won't suffice anymore.
+The more lines are touched by an individual commit, the more lines should be
+used to explain the change.
+
+The project implements lints using github actions to meet such criteria.
```
Comment (TheNeikos) [inline CONTRIBUTING.md:68]:
```suggestion
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -33,20 +36,87 @@ When possible, fill out as much detail in the pull request template as is
...
+
+Also, some hard criteria is checked via github action lints:
```
Comment (TheNeikos) [inline CONTRIBUTING.md:70]:
```suggestion
Also, some hard criteria are checked via github action lints:
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -33,20 +36,87 @@ When possible, fill out as much detail in the pull request template as is
...
+* The `Signed-off-by` trailer lint must be present in all commits (also see
+section "Ensuring the Signed-off-by-trailer")
+* The commit body must meet the default formatting of a commit message, that
+is
+* Subject line not longer than 50 characters
+* Subject capitalized
+* No period at the end of the subject
+* Imperative mood in the subject
+* Second line empty
+* Body lines not longer than 72 characters
```
Comment (TheNeikos) [inline CONTRIBUTING.md:81]:
```suggestion
    * Subject line not longer than 50 characters
    * Subject capitalized
    * No period at the end of the subject
    * Imperative mood in the subject
    * Second line empty
    * Body lines not longer than 72 characters
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -33,20 +36,87 @@ When possible, fill out as much detail in the pull request template as is
...
+
+Commits using the `--fixup` or `--squash` feature of `git-commit` are allowed,
+but won't be merged (read more about this in the "merge strategies" section).
```
Comment (TheNeikos) [inline CONTRIBUTING.md:84]:
```suggestion
but will have to be squashed into the history before being merged.
Read more about this in the "merge strategies" section.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -33,20 +36,87 @@ When possible, fill out as much detail in the pull request template as is
...
+### Ensuring the Signed-off-by-trailer
+
+Contributors need to sign the "Contributor License Agreement". They do so by
+signing off each individual commit with a so-called "Signed-off-by Trailer".
+Git offers the `--signoff`/`-s` flags when using `git-commit` to commit
+changes.
+
+To ensure that all commits have the trailer present in the commit message, a
+CI lint is installed in the github actions workflows of the project. This lint
+blocks pull requests to be merged if one or more of the commits of the pull
```
Comment (TheNeikos) [inline CONTRIBUTING.md:95]:
```suggestion
blocks merging pull requests if one or more of the commits of the pull
```

or 

```suggestion
blocks pull requests from being merged if one or more of the commits of the pull
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -59,19 +129,73 @@ used. These fixup commits should then be squashed (usually `git rebase -i
...
+developed over a long period might contain merges. They might even merge
+`master`/`main` to get up-to-date with the latest developments. Though it is
+not encouraged to have such long-running pull requests and discouraged to
+merge `master`/`main` into a PR branch, the project acknowledges that
+sometimes it is not avoidable.
+
+To summarize, the following requirements have to be met before a pull request
+can be merged:
+
+* Reviews of the relevant persons (ensured via github setting)
```
Comment (TheNeikos) [inline CONTRIBUTING.md:165]:
```suggestion
* Reviews of the relevant people (ensured via github setting)
```

**Review by TheNeikos — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 6 above)
Comment (matthiasbeyer) [reply CONTRIBUTING.md:84]:
```suggestion
but will have to be squashed by the pull request author before the being merged.
```

Comment (TheNeikos) [reply CONTRIBUTING.md:83]: remove 'the' and that's good

**Comment by matthiasbeyer** [comment]
Squashed all cleanup patches.

**Review by TheNeikos — APPROVED** [review]

**Review by didier-wenzek — DISMISSED** [review]
We need to discuss the [commit done with the false assumption there is no test team](https://github.com/thin-edge/thin-edge.io/pull/1384/commits/115d23af7811a49f088534a9645601b6de373312)

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -33,12 +33,6 @@ When possible, fill out as much detail in the pull request template as is
 reasonable. Most important is to reference the GitHub issue that you are
 addressing with the PR.
 
-**NOTE:** GitHub has [a feature](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
-that will automatically close issues referenced with a keyword (such as "Fixes")
-by a PR or commit once the PR/commit is merged. Don't use these keywords. We
-don't want issues to be automatically closed. We want our testers to
-independently verify and close them.
```
[the remark this answers is not in the document]
Comment (matthiasbeyer) [reply CONTRIBUTING.md:40]: Okay, I will push a revert for this! :+1:

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -38,10 +38,50 @@ addressing with the PR.
 
 ## Writing good commit messages
 
-Git commit messages should explain the how and why of your change and be
-separated into a brief subject line followed by a more detailed body. When in
-doubt, follow this guide for good commit messages and you can’t go wrong:
-[https://chris.beams.io/posts/git-commit/](https://chris.beams.io/posts/git-commit/).
```
[the remark this answers is not in the document]
Comment (matthiasbeyer) [reply CONTRIBUTING.md:47]: Feel free to resolve this if you think [the fixup](https://github.com/thin-edge/thin-edge.io/pull/1384/commits/93056a070969d8a1f47a363ec35f62aaf5782a0f) is okay.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -41,12 +44,87 @@ independently verify and close them.
...
+* The commit body must meet the default formatting of a commit message, that
+is
+    * Subject line not longer than 50 characters
+    * Subject capitalized
+    * No period at the end of the subject
+    * Imperative mood in the subject
+    * Second line empty
+    * Body lines not longer than 72 characters
+
+You can also read about commit message in
```
Comment (TheNeikos) [inline CONTRIBUTING.md:88]:
```suggestion
You can also read about commit messages in
```

**Review by rina23q — APPROVED** [review]
LGTM

**Comment by matthiasbeyer** [comment]
Squashed away the fixup patches.

**Comment by matthiasbeyer** [comment]
@rina23q All review points addressed as far as I can see. Feel free to merge now! :tada:

File: CONTRIBUTING.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (didier-wenzek) [reply CONTRIBUTING.md]: Thank you,

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**